### ISSUE TITLE
Syntax error in agent creation with 'type' in python 3.9

### ISSUE THREAD

**Issue by meissnereric** [cpacker/MemGPT#664]
**Describe the bug**

I think the use of 'type' (which is a special keyword in Python) is giving issues here. It could be something else, but this code runs fine on my machine in Python 3.11 so I'm just assuming something changed between those two versions.

Autogen version 0.2.0

```
Traceback (most recent call last):
  File "/home/.../game_example/memgpt_example.py", line 15, in <module>
    from memgpt.autogen.memgpt_agent import create_memgpt_autogen_agent_from_config
  File "/home/.../lib/python3.9/site-packages/memgpt/autogen/memgpt_agent.py", line 218
    match type:
          ^
```

I imagine just changing the name of 'type' to something not special will fix it?

**Please describe your setup**

- [X ] MemGPT version
  - 0.2.7
- [X ] How did you install memgpt?
  - `pip install pymemgpt`? 
- [x] Describe your setup
  - WSL, running memgpt in python (using Autogen to be clear) not through CLI.

**Comment by cpacker** [comment]
can you add your autogen version? `pip list | grep pyautogen`

**Comment by meissnereric** [comment]
Sure added. 0.2.0

**Comment by cpacker** [comment]
I think this might have to do with the fact that match-case was introduced in Python 3.10: https://docs.python.org/3/whatsnew/3.10.html

So upgrading your Python should fix the problem. I'll make a PR right now anyways that reverts off of match-case so that you can still use Python 3.9.

**Comment by cpacker** [comment]
@meissnereric this should be fixed by PR #691 (which will be part of the upcoming 0.2.10 release). Should also be fixable by upgrading your python to python >= 3.10.
